iib/workers/tasks/build_regenerate_bundle.py

```diff
@@ -529,3 +540,31 @@ def _apply_repo_enclosure(bundle_metadata, org_enclose_repo_namespace, org_enclo
         replacement_pullspecs[pullspec] = new_pullspec
 
     _replace_csv_pullspecs(bundle_metadata, replacement_pullspecs)
+
+
+def _get_related_bundle_images(bundle_metadata, request_id):
+    """
+    Get bundle images in the CSV files of the bundle being regenerated and store them in a file.
+
+    :param dict bundle_metadata: the dictionary of CSV's and relatedImages pull specifications
+    :param int request_id: the ID of the IIB build request
+    """
+    worker_config = get_worker_config()
+    related_bundles_dir = worker_config['iib_request_related_bundles_dir']
+    related_bundles_file = os.path.join(related_bundles_dir, f'{request_id}_related_bundles.json')
+
+    if os.path.isfile(related_bundles_file):
+        log.debug('Related bundles already exist for this request. Skipping...')
+        return
+
+    related_bundle_images = []
+    for related_pullspec_obj in bundle_metadata['found_pullspecs']:
+        related_pullspec = related_pullspec_obj.to_str()
+        if yaml.load(
+            get_image_label(related_pullspec, 'com.redhat.delivery.operator.bundle') or 'false'
```

Comment:
I'm confused about this. Why are we only providing the related images that are themselves bundles? Don't we need to provide a list of all relatedImages a bundle specified?

Reply:
Nope, we only need relatedImages which are themselves bundles. This is for meta-operators which install other operators.

Reply:
Thanks for the clarification. We need to make this much clearer in the code and docs.

Reply:
Could we maybe comment the code?

Reply:
The function definition mentions that it finds bundle images in the CSVs. Do we still need a comment here?

Reply:
@yashvardhannanavati, I don't think it's _needed_ but I think we have a better chance of avoiding future confusion if we add something like this:
```python
# Important: not all related images are bundle images!
if yaml.load(...
```
But this is borderline paranoia so use your best judgement to accept or ignore this suggestion.